core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -199,6 +199,7 @@ class JobData private[spark](
     val completionTime: Option[Date],
     val stageIds: collection.Seq[Int],
     val jobGroup: Option[String],
+    val jobTags: collection.Seq[String],
```

Comment:
I wonder if the following would fix the mima check
```
class JobData private[spark](
    val jobId: Int,
    val name: String,
    val description: Option[String],
    val submissionTime: Option[Date],
    val completionTime: Option[Date],
    val stageIds: collection.Seq[Int],
    val jobGroup: Option[String],
    val status: JobExecutionStatus,
    val numTasks: Int,
    val numActiveTasks: Int,
    val numCompletedTasks: Int,
    val numSkippedTasks: Int,
    val numFailedTasks: Int,
    val numKilledTasks: Int,
    val numCompletedIndices: Int,
    val numActiveStages: Int,
    val numCompletedStages: Int,
    val numSkippedStages: Int,
    val numFailedStages: Int,
    val killedTasksSummary: Map[String, Int],
    val jobTags: Option[collection.Seq[String]])
```

Reply:
I can try, though it would still be an incompatible constructor...

Reply:
Discussed with @hvanhovell . This is a private[spark] constructor, this doesn't need MIMA compatibility.
https://github.com/apache/spark/commit/fb67ce26800119a01633d5ffb05d8e341f76d3e6#diff-23d5115b4b5d50306b2908ca9bde032dc88f57d84da0316ddef146681446bd3aR93 analogously changes StageData constructor and excludes it. will add to MimaExcludes.

Reply:
@juliuszsompolski OK. We will need to check the backward compatibility of Spark history server as well.
cc @rednaxelafx

Reply:
@gengliangwang @rednaxelafx could you help what check could I make?
The change is a new optional (repeated) field in the proto. Any older version of SHS reading that proto will ignore it. Any new version of SHS reading proto from an older version will see it as an empty Seq (since empty repeated == empty seq). So it should be backwards and forwards compatible. Are there tests that I could add for that? I consulted https://github.com/apache/spark/commit/fb67ce26800119a01633d5ffb05d8e341f76d3e6 which did a similar thing (add new fields to StageData and don't see extra tests that I could add.

Reply:
@juliuszsompolski Yes I meant the backward compatibility of Spark event logs which is in JSON format. I do remember that new fields will be ignored so probably it should be fine, just double confirming.